src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.cs

```diff
@@ -0,0 +1,1916 @@
+                    return ParseEndAnchor();
+                case RegexKind.BackslashToken:
+                    return ParseEscape(_currentToken, allowTriviaAfterEnd: true);
+                case RegexKind.OpenBracketToken:
+                    return ParseCharacterClass();
+                case RegexKind.OpenParenToken:
+                    return ParseGrouping();
+                case RegexKind.CloseParenToken:
+                    return ParseUnexpectedCloseParenToken();
+                case RegexKind.OpenBraceToken:
+                    return ParsePossibleUnexpectedNumericQuantifier(lastExpression);
+                case RegexKind.AsteriskToken:
+                case RegexKind.PlusToken:
+                case RegexKind.QuestionToken:
+                    return ParseUnexpectedQuantifier(lastExpression);
+                default:
+                    return ParseText();
+            }
+        }
+
+        private RegexPrimaryExpressionNode ParsePossibleUnexpectedNumericQuantifier(RegexExpressionNode lastExpression)
+        {
+            // Native parser looks for something like {0,1} in a top level sequence and reports
+            // an explicit error that that's not allowed.  However, something like {0, 1} is fine
+            // and is treated as six textual tokens.
+            var openBraceToken = _currentToken.With(kind: RegexKind.TextToken);
+            var start = _lexer.Position;
+
+            if (TryParseNumericQuantifierParts(
+                    out _, out _, out _, out _))
+            {
+                // Report that a numeric quantifier isn't allowed here.
+                CheckQuantifierExpression(lastExpression, ref openBraceToken);
+            }
+
+            // Started with { but wasn't a numeric quantifier.  This is totally legal and is just
+            // a textual sequence.  Restart, scanning this token as a normal sequence element.
+            ResetToPositionAndConsumeCurrentToken(start, allowTrivia: true);
+            return new RegexTextNode(openBraceToken);
+        }
+
+        private RegexPrimaryExpressionNode ParseUnexpectedCloseParenToken()
+        {
+            var token = _currentToken.With(kind: RegexKind.TextToken).AddDiagnosticIfNone(
+                new EmbeddedDiagnostic(WorkspacesResources.Too_many_close_parens, _currentToken.GetSpan()));
+
+            // Technically, since an error occurred, we can do whatever we want here.  However,
+            // the spirit of the native parser is that top level sequence elements are allowed
+            // to have trivia.  So that's the behavior we mimic.
+            ConsumeCurrentToken(allowTrivia: true);
+            return new RegexTextNode(token);
+        }
+
+        private RegexPrimaryExpressionNode ParseText()
+        {
+            // Allow trivia between this piece of text and the next sequence element
+            return new RegexTextNode(ConsumeCurrentToken(allowTrivia: true).With(kind: RegexKind.TextToken));
```

Comment:
Is it possible for the token to have a `Value`? Should it be cleared when turned into a text token?

Reply:
It shoudln't.  We only parse out special tokens (i.e. names/numbers) in contexts where we know we'll use htem.  ParseText is only part of our main loop where we're getting normal tokens (which shoudl not have values).  I'm going to add an assert as such.

Reply:
#resolvd